doc/source/whatsnew/v0.24.0.txt

```diff
@@ -83,6 +83,7 @@ Other Enhancements
 - :func:`read_html` copies cell data across ``colspan``s and ``rowspan``s, and it treats all-``th`` table rows as headers if ``header`` kwarg is not given and there is no ``thead`` (:issue:`17054`)
 - :meth:`Series.nlargest`, :meth:`Series.nsmallest`, :meth:`DataFrame.nlargest`, and :meth:`DataFrame.nsmallest` now accept the value ``"all"`` for the ``keep` argument. This keeps all ties for the nth largest/smallest value (:issue:`16818`)
 - :class:`IntervalIndex` has gained the :meth:`~IntervalIndex.set_closed` method to change the existing ``closed`` value (:issue:`21670`)
+- :func:`~DataFrame.to_csv` and :func:`~DataFrame.to_json` now support ``compression='infer'`` to infer compression based on filename (:issue:`15008`)
```

Comment:
csv, json, pickle as well?

Reply:
and this is for both reading and writing?

Reply:
* Not sure about `pickle`, as it wasn't tested in the original diff.  Can check.
* Nope, just for writing.  Reading already has support (why we didn't support it for both reading and writing in the first place befuddles me a little...)

Reply:
Ha, `to_pickle` already has support for `infer` per the docs.  Now I'm really confused how we didn't have this for `to_csv` and `to_json`...